### PR TITLE
(deprecated) stardrop: init at 1.2.1

### PR DESCRIPTION
**Pull request by sofiedotcafe**
## Description of changes

Stardrop is an open-source, cross-platform mod manager for the game Stardew Valley. 
https://floogen.gitbook.io/stardrop/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/st/stardrop/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  zip,
+}:
+
+buildDotnetModule rec {
+  pname = "stardrop";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "Floogen";
+    repo = "${pname}";
```
Comment (eclairevoyant) [inline pkgs/by-name/st/stardrop/package.nix:15]:
```suggestion
    repo = "stardrop";
```

File: pkgs/by-name/st/stardrop/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  projectFile = "Stardrop/Stardrop.csproj";
+  buildInputs = [ zip ];
+
+  dotnet-sdk = dotnetCorePackages.sdk_7_0;
+  dotnet-runtime = dotnetCorePackages.runtime_7_0;
+
+  nugetDeps = ./deps.nix;
+
+  meta = {
+    description = "Stardrop is an open-source, cross-platform mod manager for the game Stardew Valley";
```
Comment (eclairevoyant) [inline pkgs/by-name/st/stardrop/package.nix:33]:
```suggestion
    description = "Open-source, cross-platform mod manager for the game Stardew Valley";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -12743,6 +12743,8 @@ with pkgs;
 
   sqls = callPackage ../applications/misc/sqls { };
 
+  stardrop = callPackage ../by-name/st/stardrop/package.nix { };
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:12746]:
```suggestion
```

File: pkgs/by-name/st/stardrop/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    rev = "v${version}";
+    hash = "sha256-VN0SrvBT5JUNraeh6YyRhcnoOl+mOB2/zk/rQeJidI8=";
+  };
+
+  patches = [
+    ./patches/csproj-for-nix.patch
+  ];
+
+  projectFile = "Stardrop/Stardrop.csproj";
+  buildInputs = [ zip ];
```
Comment (eclairevoyant) [inline pkgs/by-name/st/stardrop/package.nix:25]:
```suggestion
```

**Comment by jh-devv** [comment]
I have not yet pushed the changes but I have added an FHS environment to the package. This was recommended on the Matrix channel since it launches a game.

**Comment by SigmaSquadron** [comment]
> I have not yet pushed the changes but I have added an FHS environment to the package. This was recommended on the Matrix channel since it launches a game.

I saw the FHS you posted on Matrix. My only nit with it is that you should avoid using the `pname` variable everywhere, and hardcode executable names instead.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (eclairevoyant) [reply pkgs/top-level/all-packages.nix:12746]: @jh-devv Please don't mark feedback as resolved if they're unresolved.

Comment (jh-devv) [reply pkgs/top-level/all-packages.nix]: > @jh-devv Please don't mark feedback as resolved if they're unresolved.

Oh, sorry, GitHub web had some issues and I couldn't commit the changes. I had to make it an "new commit"

-- Fixed with #b301e6b

**Review by eclairevoyant — COMMENTED** [review]
Does this really not work without an FHS env? We're building the thing from source here, surely we can patch as needed

**Comment by eclairevoyant** [comment]
Also, please rebase and squash everything into the first commit

**Comment by jh-devv** [comment]
> Also, please rebase and squash everything into the first commit

Yikes, didn't mean to do that, sorry!

**Comment by jh-devv** [comment]
> Does this really not work without an FHS env? We're building the thing from source here, surely we can patch as needed

Haven't looked into the code but I suspect heavily it does not.

**Comment by eclairevoyant** [comment]
> I suspect heavily it does not.

Please try. We really want to avoid FHS envs.

**Comment by jh-devv** [comment]
> > I suspect heavily it does not.
> 
> Please try. We really want to avoid FHS envs.

Oh, well, I'll take a look at the code! :+1:

**Comment by jh-devv** [comment]
Without FHS, the applications works, you can't run the game though. I suspect this is because the game is linked against FHS paths. 

```
❯ readelf -a Stardew\ Valley | grep "/" 

  OS/ABI:                            UNIX - GNU
      [Requesting program interpreter: /lib64/ld-linux-x86-64.so.2]
  Tag        Type                         Name/Value
 0x000000000000000f (RPATH)              Library rpath: [$ORIGIN/netcoredeps]
 ```

You can really patch the game, without the updates braking and stuff like that.

**Comment by eclairevoyant** [comment]
The FHS env is only going to apply to the application being built here, how is the game relevant?

**Comment by jh-devv** [comment]
> The FHS env is only going to apply to the application being built here, how is the game relevant?

Because, it is an *mod manager and a launcher* for the game. You can click a button to launch the game.

**Comment by jh-devv** [comment]
You could launch it from steam (since, it just manages the mod files) and SMAPI (mod loader) is already patched in. But then you lose functionality like better logs, priority mods etc

**Comment by eclairevoyant** [comment]
> _and a launcher_ for the game.

This was not mentioned at all in the original description. Now this whole rigmarole makes sense.

**Comment by jh-devv** [comment]
> > _and a launcher_ for the game.
> 
> This was not mentioned at all in the original description. Now this whole rigmarole makes sense.

Oh, yeah, sorry.... but yeah... You don't have to use it as a launcher, but you can. I for example, don't.

**Comment by poperigby** [comment]
Anything else need to be done to get this merged?

**Comment by jh-devv** [comment]
Silly me :3 Accidentally deleted the branch lol, it's back up now!

**Comment by jh-devv** [comment]
Some problems with the fork, i'll get it back this evening!

**Pull request closed without merging**